snapcraft.yaml

```diff
@@ -0,0 +1,23 @@
+name: konf
+base: core18
+version: '0.1'
+summary: K8s config generator
+description: K8s config generator
+grade: devel
+confinement: devmode
```

Comment:
I am not sure about this but why it can be a `strict` one?

Reply:
As it stands I think we need `classic` or `devmode` otherwise we can't read the config yaml from the filesystem. If we want to move the configuration files into the repos there is no way we can bundle the configurations in the snap :(

Reply:
This actually raises another point, if it needs `classic` it might need manual approval to be released, do we still want a snap if that's the case? @tbille @nottrobin @carkod

Reply:
i think you only need `classic` approval once - when it's approved it will stay approved and we can do releases without subsequent approvals. So even if we do need `classic`, I don't think this should really impact on our plans.

Making it `devmode` (which would also force us to use the `beta` risk level) is a workable hack in the meantime. It just means that the snap won't auto-refresh in the background, and we'd have to manually refresh it with `snap refresh --devmode` to pull down the latest version. This isn't really an issue for us, as we'll definitely be wanting to explicitly run `snap refresh` in our Jenkins jobs to ensure we're on the latest version anyway.

In the long run, though, `devmode` is gonna be a pain. Mostly it will mean that we can't expect local installations of `konf` on our local machines to stay up to date automatically.

However, the snap store guardians will want to see that we've exhausted all options for making this a `strict` snap before approving `classic`.

We can give the snap permission to access all files in the home directory with [the `home` interface](https://snapcraft.io/docs/home-interface), but I guess the problem is that we need the snap to access files in `/var/lib/jenkins` for running in Jenkins - is that what you meant @jpmartinspt? The best discussion of this issue I could find is [here](https://forum.snapcraft.io/t/how-can-i-use-snap-when-i-dont-use-home-user/3352/50). I think the best solution right now would be to ask IS to configure our Jenkins installation to map the Jenkins home directory to `/home/jenkins`, which would allow `strict` snaps access to all files in there with the `home` interface. And seems like a nice structure anyway.

What do you think is the best way forward?

Reply:
Oh also, note that a `devmode` snap and a `classic` snap behave very differently, and so don't expect you can simply change from `devmode` to `classic` and expect it to just work.

A `devmode` snap basically runs the same as a `strict` snap, but with the confinement rules removed. This means, among other things, that `$PATH` and `$HOME` point to snap-specific locations. Whereas a `classic` snap inherits all the user's normal environment variables. This makes installing and working with snap dependencies rather confusing and non-intuitive. You'll see what I mean when you try it ;)